codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -649,12 +649,17 @@ final void takeChild(DefaultStream child, boolean exclusive, List<ParentChangedE
             if (exclusive && !children.isEmpty()) {
                 // If it was requested that this child be the exclusive dependency of this node,
                 // move any previous children to the child node, becoming grand children of this node.
-                for (DefaultStream grandchild : retain(child).values()) {
-                    child.takeChild(grandchild, false, events);
+                Iterator<PrimitiveEntry<DefaultStream>> itr = retain(child).entries().iterator();
```

Comment:
nit: we really should consider renaming the `retain` method.  I was scratching my head for a while trying to understand this code :).  Not something we need to do now, but maybe something like `removeAllChildrenExcept` or `removeAllOtherChildren`.